### ISSUE TITLE
Investigating Monkeyrunner for functional tests

### ISSUE THREAD

**Issue by isabelrios** [mozilla-magnet/magnet-client#71]
Started to investigate how to use monkeyrunner for android.
I got a simple test to install the app both in a real device and in the emulator.
But if the idea is to have the tests running on the emulator there is the problem that it does not support bluetooth, so we could only simulate a limited tests like to install the app, see the message that bluetooth is needed and unistall it.
The only option I see is to run the tests in a real device...but I would probably be missing more options :)
cc @samgiles

**Comment by samgiles** [comment]
Hmm, that's an issue, we would miss out on testing the bluetooth discovery automatically, but, we could test the UI using mDNS though.

**Comment by isabelrios** [comment]
Sorry, my first comment was not totally correct, I got to install the app in the emulator but it was closed a few seconds later with the message: Unfortuntely, magnet has stopped. Thought was something related to my emulator, but it is not, it is related to the blueetooth not being supported by the emulator.
I tried modifying a little bit the app, removing the blutooth permission and so and now I get the app open, it is not closed, but with errors of course (I already run adb reverse tcp:8081 tcp:8081).
![screenshot_20160513-151018](https://cloud.githubusercontent.com/assets/1897507/15250253/2b4bf3ca-1924-11e6-901a-e88672b0b86a.png)

So in case we want to have some tests on the emulator, all the bluetooth part has to be modified :S Not sure if we can do all using mDNS (sorry not very familiar with that) @samgiles

**Comment by arcturus** [comment]
:( yeah, just read that the emulator doesnt support BT (among other things, like wifi)

http://developer.android.com/tools/devices/emulator.html#starting

**Comment by samgiles** [comment]
@isabelrios Incidentally, there should  be a fix for that particular error you're seeing in #80 

We can add probably add to the config file on a build to have tiles show up by default. https://github.com/mozilla-magnet/magnet-client/blob/master/config.js#L23

**Comment by isabelrios** [comment]
@samgiles Thanks! I will try again and see if with that at least we can open the app via emulator without crashes, then adding some tiles by default would definitely help to start testing the builds on the emulator.

**Comment by isabelrios** [comment]
Hi @samgiles, some progress here. 
As you said, that error is not shown anymore. And, changing the conf file, I can see the defaults tiles on the emulator.
But, since the app keeps scanning, searching for beacons, after some seconds in which I see the default tiles
![default tiles](https://cloud.githubusercontent.com/assets/1897507/15537193/42a9b142-2275-11e6-9be1-868d10cb2fad.png)

I got this error:
![emulator stopscanerror](https://cloud.githubusercontent.com/assets/1897507/15537113/e1d07de2-2274-11e6-96b7-db3efed0a393.png)

Should it be possible to force stop the scanning? 
Thanks!

**Comment by isabelrios** [comment]
@samgiles cool! with that latest commit that error is not shown and seems that the app does not crash! have to do a few more tests but this looks good
Thanks!

**Comment by isabelrios** [comment]
Closing this issue since finally I got UI tests running using Monkeyrunner. These are valid and can be used in CI once we could build a release build with fixed tiles (issue #166 ) In this [PR](https://github.com/mozilla-magnet/magnet-client/pull/132) I got them launched in the emulator but since we do not have the UI is not possible to incorporate them yet.

Also, after further investigations seems that we could get better UI tests with Appium both in Android and iOS real devices. Although on the emulator only Android tests would be possible. These tests are implemented and working locally so far, can be found [here](https://github.com/isabelrios/appiumTests). Adding them (the android ones) to CI would be easy but the same than for Monkeyrunner tests, we would need kind of a test build (release build but with fixed tiles)
